File: Games/Uzta/src/main/java/jokatu/game/games/echo/game/UztaFactory.java
package jokatu.game.games.echo.game;

import jokatu.game.AbstractGameFactory;
import jokatu.game.GameID;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class UztaFactory extends AbstractGameFactory<Uzta> {

	@NotNull
	@Override
	protected Uzta produce(@NotNull GameID gameID, @NotNull String creatorName) {
		return new Uzta(gameID);
	}
}

File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/game/UztaFactory.kt
package jokatu.game.games.uzta.game

import jokatu.game.AbstractGameFactory
import jokatu.game.GameID
import org.springframework.stereotype.Component

@Component
class UztaFactory : AbstractGameFactory<Uzta>() {
	override fun produce(gameID: GameID, creatorName: String): Uzta {
		return Uzta(gameID)
	}
}
